src/uu/sort/src/sort.rs

```diff
@@ -87,13 +99,165 @@ impl Default for Settings {
             compare_fns: Vec::new(),
             transform_fns: Vec::new(),
             salt: String::new(),
+            selectors: vec![],
+            separator: None,
+        }
+    }
+}
+
+#[derive(PartialEq, Debug)]
+struct Token {
+    // from (inclusive)
+    from: usize,
+    // to (exclusive)
+    to: usize,
+}
+
+struct Line {
+    // If None, this line was not transformed and string is still the original
+    original: Option<String>,
+    string: String,
+    tokens: Option<Vec<Token>>,
+}
+
+impl Line {
+    fn new(input: String, settings: &Settings) -> Self {
+        let tokens = if !settings.selectors.is_empty() {
+            Some(tokenize(&input, settings.separator))
+        } else {
+            // If there are no selectors we will never need tokens, so don't calculate them in this case.
+            None
+        };
+        let transformed = transform(&input, settings);
+        let string;
+        let original;
+        if let Some(transformed) = transformed {
```

Comment:
please use tuples
https://doc.rust-lang.org/rust-by-example/primitives/tuples.html
to have
let (string, original) = if ...

Reply:
Is this `destructuring_assingment`? I thought that wasn't stable yet (https://github.com/rust-lang/rust/issues/71126), but I'll try once I'm back to my computer

Reply:
Oh, I might be wrong :)

Reply:
No, looks like you're right :)

Reply:
Cool, this is rare :)